### PR TITLE
Some tweaks to the Client

### PR DESCRIPTION
**Pull request by abidlabs**
This PR:

* Fixes a small typo in `Client.view_api(all_endpoints=True)`
* Brings back underscores in parameter names when displaying API info (Closes: #4221)
* Moves the state-removal logic outside of serialization (Closes: #4081) and adds a test
* Fixes a separate bug, which is that state-removal logic wouldn't be applied to generators
* Releases a new version of the Client, since we likely (hopefully) won't work on it for the rest of the 3.x milestone

### CODE REVIEW

File: client/python/test/test_client.py [hunk 1]
```diff
@@ -308,6 +308,18 @@ def test_upload_file_upload_route_does_not_exist(self):
                 client.submit(1, "foo", f.name, fn_index=0).result()
                 serialize.assert_called_once_with(1, "foo", f.name)
 
+    def test_state_without_serialize(self, stateful_chatbot):
```
Comment (freddyaboulton) [inline client/python/test/test_client.py:311]: We can add a `serialize` argument to `connect` to use it in this test.

**Review by freddyaboulton — APPROVED** [review]
Thanks for the fix @abidlabs ! Confirmed it fixes the original issue

File: client/python/test/test_client.py (hunk 1 above)
Comment (abidlabs) [reply client/python/test/test_client.py:311]: Agreed, that would be better!

Comment (abidlabs) [reply client/python/test/test_client.py:309]: Fixed

**Comment by abidlabs** [comment]
🧡

**Pull request merged**